index.js

```diff
@@ -11,8 +11,26 @@ const COLORS = 'https://nt-cdn.s3.amazonaws.com/colors.json';
  * @returns Promise
  */
 const fetchColors = ({ name, hex, compName, compHex }) => {
-  throw Error('Not implemented');
+  console.log(compName);
+  //Fetches the Color Object from the API defined above.
+  return fetch(COLORS)
+    // The promise is returned and converted to json.
+    .then(res => res.json())
+    // The Json result is filtered by the parameters that are passed in the function. Then helps to ensure the promise is completed before the filtering code is applied. 
+    .then(colorObj => {
+      const filteredColor = colorObj.filter(color =>
+        (name === undefined || (color.name.toLowerCase().includes(name.toLowerCase()))) &&
```

Comment:
Remember to DRY your code. The string comparison could be moved to its own function so if the requirement changed to be case sensitive you only need to do it in one place.